testcases/MachineConfig.py

```diff
@@ -56,16 +55,42 @@ def setUp(self):
             self.system_name = conf.args.system_name
             self.cv_HMC = self.cv_SYSTEM.hmc
             self.lpar_prof = conf.args.lpar_prof
+            self.secureboot_valid_states = ['on', 'off']
         else:
             self.skipTest("Functionality is supported only on LPAR")
 
+    def check_secureboot_state(self):
+        '''
+        Checks the validity of the states, set as parameter in the configuration file.
+        Valid states are either 'on' or 'off'.
+        Return : 
+        True  - if secureboot is set to 'on'
+        False - if secureboot is set to 'off' 
+        '''
+        sb_valid_state = re.findall("secureboot=[a-z][a-z]+", str(self.machine_config))[0].split('=')[1]
+        if str(sb_valid_state) not in self.secureboot_valid_states:
+            self.skipTest("%s is not a valid state for secureboot. "
+                          "Secureboot valid states can be either set to 'on' or 'off' ")
+        if sb_valid_state == 'on':
+            return True
+        else:
+            return False
+
     def runTest(self):
         status=0
         if self.machine_config.__contains__('lpar'):
+            config_value = self.machine_config['lpar']
             status=LparConfig(self.cv_HMC,self.system_name,self.lpar_name,self.lpar_prof,self.machine_config['lpar']).LparSetup()
             if status:
                 self.fail(status)
 
+            if 'secureboot' in config_value:
+                enable = self.check_secureboot_state()
+                status = LparConfig(self.cv_HMC, self.system_name, self.lpar_name,
+                                  self.lpar_prof, self.machine_config['lpar']).hmc_secureboot_on_off(enable=enable)
```

Comment:
have this in [LparSetup](https://github.com/open-power/op-test/blob/7f712f557475b9f076e16e800f68c64f49f0ab67/testcases/MachineConfig.py#L181C9-L181C18)  you can inspire from other lpar settings like 
if 'secureboot' in config_value:
.hmc_secureboot_on_off(enable=enable)